addon/components/ember-table.js

```diff
@@ -0,0 +1,564 @@
+    // we need to wait for the table to be fully rendered before antiscroll can
+    // be used
+    return Ember.run.next(this, this.updateLayout);
+  },
+
+  tableWidthNowTooSmall: function() {
+    if ((this.get('_state') || this.get('state')) !== 'inDOM') {
+      return false;
+    }
+    var oldTableWidth = this.get('_width');
+    var newTableWidth = this.$().parent().width();
+    // TODO(azirbel): This should be 'columns', I believe. Fix separately.
+    var totalColumnWidth = this._getTotalWidth(this.get('tableColumns'));
+    return (oldTableWidth > totalColumnWidth) && (newTableWidth < totalColumnWidth);
+  },
+
+  updateLayout: function() {
+    if ((this.get('_state') || this.get('state')) !== 'inDOM') {
+      return;
+    }
+    // updating antiscroll
+    this.$('.antiscroll-wrap').antiscroll().data('antiscroll').rebuild();
+    if (this.get('columnsFillTable')) {
+      return this.doForceFillColumns();
+    }
+  },
+
+  // Iteratively adjusts column widths to adjust to a changed table width.
+  // Attempts to scale columns proportionally. However, if a column hits a min
+  // or max width after scaling proportionally, we need to respect that setting.
+  // In that case, keep iterating until all column widths are set to the best
+  // they can be. Note that this may fail to arrive at the table width if the
+  // resizable columns are all restricted by min/max widths.
+  doForceFillColumns: function() {
+    var allColumns = this.get('columns');
+    var columnsToResize = allColumns.filterProperty('canAutoResize');
+    var unresizableColumns = allColumns.filterProperty('canAutoResize', false);
+    var availableWidth = this.get('_width') - this._getTotalWidth(unresizableColumns);
+    var doNextLoop = true;
+    var nextColumnsToResize = [];
+    var totalResizableWidth;
+    var newWidth;
+
+    while (doNextLoop) {
+      doNextLoop = false;
+      nextColumnsToResize = [];
+      totalResizableWidth = this._getTotalWidth(columnsToResize);
+      /*jshint loopfunc:true */
+      // TODO(azirbel): Revisit JSHint error above
+      columnsToResize.forEach(function(column) {
+        newWidth = Math.floor((column.get('width') / totalResizableWidth) * availableWidth);
+        if (newWidth < column.get('minWidth')) {
+          doNextLoop = true;
+          column.set('width', column.get('minWidth'));
+          availableWidth -= column.get('width');
+        } else if (newWidth > column.get('maxWidth')) {
+          doNextLoop = true;
+          column.set('width', column.get('maxWidth'));
+          availableWidth -= column.get('width');
+        } else {
+          column.set('width', newWidth);
+          nextColumnsToResize.pushObject(column);
+        }
+        columnsToResize = nextColumnsToResize;
+      });
+    }
+  },
+
+  onBodyContentLengthDidChange: Ember.observer(function() {
+    return Ember.run.next(this, function() {
+      return Ember.run.once(this, this.updateLayout);
```

Comment:
I feel like this should actually be a `scheduleOnce` on `afterRender`. I might be wrong though.